e2e/test/E2EMsTestBase.cs

```diff
@@ -44,6 +53,20 @@ public void TestCleanup()
             Logger.Event(TestFinishedEventName, extraProperties);
             // As this is not an application that keeps running, explicitly flushing is required to ensure we do not lose any logs.
             Logger.SafeFlush();
+
+            _listener.Dispose();
```

Comment:
Call Dispose here and move _listener.Dispose() to Dispose()

Reply:
we do want to dispose the console test listener for each test run, and not when the base test class is disposed.

Reply:
And that should happen if you call Dispose() here and just move the _listener.Dispose() to the Dispose method. Ideally if this class is IDisposable, it should dispose everything disposable in the Dispose method and not in the TestCleanup.

Reply:
> Ideally if this class is IDisposable, it should dispose everything disposable in the Dispose method and not in the TestCleanup.

I agree with this, but how would we manage the scenario where you want to dispose some resources once you are done with test1, and some resources that you want to dispose once you are done with the entire test suite.
Wouldn't managing the lifecycle of the test specific resource inside `TestCleanup` make more sense, then?

Reply:
What is the difference? The only place we are initializing the listener is in the TestInitialize. What I was suggesting is have a generic dispose method you can call and based on if something is initialized or not, it will be disposed in the TestCleanup